rpc/client.go

```diff
@@ -137,6 +137,9 @@ func (c *Client) connect(opts *retry.Options, context *Context) {
 		conn, err := tlsDialHTTP(c.addr.Network(), c.addr.String(), tlsConfig)
 		// Could be many errors: bad host; bad certificate ...
 		if err != nil {
+			if t, ok := err.(*net.OpError); ok && t.Temporary() {
+				return retry.Continue, err
```

Comment:
Might want to `s/*net.OpError/net.Error/g` instead. Which errors are temporary? The docs don't say.